source/java/src/main/java/com/uber/neuropod/Neuropod.java

```diff
@@ -0,0 +1,198 @@
+package com.uber.neuropod;
+
+import java.nio.*;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class is an one to one mapping to cpp NeuropodValue type. Need to manually call close
+ * Method after using this class
+ */
+public class Neuropod extends NativeClass {
+
+    /**
+     * Wrap a native handle.
+     *
+     * @param nativeHandle the native handle
+     */
+    public Neuropod(long nativeHandle) {
+        super(nativeHandle);
+    }
+
+    /**
+     * Load a model from the file path and use default options
+     *
+     * @param neuropodPath the neuropod path
+     */
+    public Neuropod(String neuropodPath) {
+        super.setNativeHandle(nativeNew(neuropodPath));
+    }
+
+    /**
+     * Load a model from the file path and use provided options
+     *
+     * @param neuropodPath the neuropod path
+     * @param options      the options
+     * @throws Exception the exception
+     */
+    public Neuropod(String neuropodPath, RuntimeOptions options) throws Exception {
+        RuntimeOptions.RuntimeOptionsNative nativeOptions = options.toNative();
+        super.setNativeHandle(nativeNew(neuropodPath, nativeOptions.getNativeHandle()));
+        nativeOptions.close();
+    }
+
+
+    /**
+     * Perform the inference calculation on the input data
+     *
+     * @param inputs the input data
+     * @param shape  the shape of the input data
+     * @return the inference result
+     * @throws Exception the exception
+     */
+    public NeuropodValueMap infer(Map<String, Object> inputs, Map<String, List<Long>> shape) throws Exception {
+        checkInputSpec(inputs, shape);
+        NeuropodValueMap valueMap = new NeuropodValueMap();
+        for (String key:inputs.keySet()) {
+            List<Long> curDim = shape.get(key);
+            NeuropodValue value = NeuropodValue.create(inputs.get(key), shape.get(key), this);
+            valueMap.addEntry(key, value);
+            value.close();
+        }
+        NeuropodValueMap ret = infer(valueMap);
+        valueMap.close();
+        return ret;
+    }
+
+    /**
+     * Gets name.
+     *
+     * @return the name
+     */
+    public String getName() {
+        return nativeGetName(super.getNativeHandle());
+    }
+
+    /**
+     * Gets platform.
+     *
+     * @return the platform
+     */
+    public String getPlatform() {
+        return nativeGetPlatform(super.getNativeHandle());
+    }
+
+    /**
+     * Perform the inference calculation on the input data
+     *
+     * @param inputs the inputs
+     * @return the inference result
+     */
+    public synchronized NeuropodValueMap infer(NeuropodValueMap inputs) {
```

Comment:
The input specs are checked by the C++ neuropod library. checkInputSpec function is to prevent errors caused in the process of converting Java data to to C++ data, it does not aim to check the sanity of the input data.